chat/consumers.py

```diff
@@ -60,7 +111,37 @@ async def connect(self):
             logger.error("Error adding channel to group %s: %s", self.room_group_name, e)
             await self.close(code=1011)
 
-    async def disconnect(self, close_code):
+    @database_sync_to_async
+    def get_user_by_id(self, user_id: int) -> Optional[User]:
+        """ Sync ORM call wrapped to async. """
+        try:
+            return User.objects.get(id=user_id)
+        except User.DoesNotExist:
+            return None
+
+    @database_sync_to_async
+    def get_or_create_chat_room(self, user1, user2) -> Tuple[Room, bool]:
+        """
+        Creates or retrieves a chat room between two users (MongoEngine version).
+        Since MongoEngine has no get_or_create, we implement it manually.
+        """
+        investor = user1 if user1.roles.filter(name='Investor').exists() else user2
```

Comment:
Role checking assumes 'roles' field and 'name' property exist, which may break if user model changes. Add attribute existence checks or handle AttributeError.
```
if hasattr(user1, 'roles') and user1.roles.filter(name='Investor').exists():
    investor = user1
else:
    investor = user2
```